### PR TITLE
Add element & evaluate tests

### PR DESCRIPTION
**Pull request by jkrems**
* `browser.getElement`: Throws a clean exception if the element doesn't exist
* `browser.getElementOrNull`: Same behavior as `-sync`'s `getElement`

### CODE REVIEW

File: lib/browser/element.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      switch (elements.length) {
+      case 0:
+        throw new Error('Element not found for selector: ' + selector);
+
+      case 1:
+        return elements[0];
+
+      default:
+        throw new Error(
+          'Selector .message has 3 hits on the page, assertions require unique elements');
```
Comment (EndangeredMassa) [inline lib/browser/element.js]: missing interpolation for `selector`?

Comment (jkrems) [inline lib/browser/element.js]: Erm, good point.

**Comment by jkrems** [comment]
Fixed the wrong error message.

**Comment by EndangeredMassa** [comment]
Whatever the behavior of `getElement`, I think it should be the same between `wd` and `sync`.

**Comment by jkrems** [comment]
Yeah, it's one of these "two bad choices" things: We are exposing all of `wd`'s APIs which have the pattern of `elementById(id)/elementByIdOrNull(id)`. So we have the choice between being consistent with the remaining `browser.*` methods or being consistent with `-sync`. My current proposal would be to migrate `-sync` to `getElement/getElementOrNull`. But that would be breaking change.

**Comment by jkrems** [comment]
Note: If we go for `getElementOrNull`, we could start by adding a deprecation warning to `-sync` whenever `getElement` returns null saying "Hey! In future this will throw, use getElementOrNull if you're expecting this element not to exist.".

**Comment by EndangeredMassa** [comment]
Bah, throwing an error just makes no sense.

I'd rather they be the same than not. We can migrate `sync`. A breaking change is fine. We can do a major version bump. We probably want to wrap the error that comes back from `getElement` saying that you can use `getElementOrNull` if you'd rather this not error.

**Comment by jkrems** [comment]
:+1: Throwing an error makes a lot of code nicer. E.g. `getElement('.foo').get('id')` will throw `"Element '.foo' not found, use getElementOrNull if that's expected"` instead of `"Can't call get of null"`.

Will add a graceful upgrade path change to `-sync`. For the major bump we might also think about the other issue (https://github.com/groupon/testium/issues/90).

**Comment by EndangeredMassa** [comment]
Sure, but the other side is more cumbersome. If you consider not finding an element to be valid, you now have to wrap in a try-catch. I suppose that's why we should have both methods.

**Comment by jkrems** [comment]
https://github.com/testiumjs/testium-driver-sync/pull/7 - Add a warning to the current behavior and add `getElementOrNull` so there's a clean migration path.

**Comment by jkrems** [comment]
Started the migration of sync, `tesitum@3.3.0` will warn when someone uses `getElement` and it returns null and will nudge them towards `getElementOrNull`.

**Comment by jkrems** [comment]
Dirty merge with master because I didn't want to mess up the rebase of `jk-the-rest` for #4.

**Comment by jkrems** [comment]
I think it will be more productive to just continue with #4 directly.

**Pull request closed without merging**